demos/serverless/src/handlers.js

```diff
@@ -312,4 +313,27 @@ function addSignalMetricsToCloudWatch(logMsg, meetingId, attendeeId) {
       putMetric(metricName, metricValue, meetingId, attendeeId);
     }
   }
+}
+
+function addEventIngestionMetricsToCloudWatch(log, meetingId, attendeeId) {
+  const putMetric =
+    metricScope(metrics => (metricName, metricValue, meetingId, attendeeId) => {
+      metrics.setProperty('MeetingId', meetingId);
+      metrics.setProperty('AttendeeId', attendeeId);
+      metrics.putMetric(metricName, metricValue);
+    });
+  const { logLevel, message } = log;
+  let errorMetricValue = 0;
+  let retryMetricValue = 0;
+  let ingestionTriggerSuccessMetricValue = 0;
+  if (logLevel === 'ERROR') {
+    errorMetricValue = 1;
+  } else if (logLevel === 'WARN' && message.match(/Retry (count)? limit reached/g)) {
+    retryMetricValue = 1;
+  } else if (message.includes('send successful')) {
+    ingestionTriggerSuccessMetricValue = 1;
+  }
+  putMetric('EventIngestionTriggerSuccess', ingestionTriggerSuccessMetricValue, meetingId, attendeeId);
+  putMetric('EventIngestionError', errorMetricValue, meetingId, attendeeId);
+  putMetric('EventIngestionRetryCountLimitReached', retryMetricValue, meetingId, attendeeId);
```

Comment:
So we want to report 3 metrics for every log that we get from events ingestion?

Reply:
So for event reporting canary run, it will POST event ingestion logs and hence I have to check the log message to know which is the type and accordingly metric that. As next steps, I would add alarm on the sum of these metrics based on a certain period say 15mins.

Reply:
Dont we only need to alarm if we see `EventIngestionError` or `EventIngestionRetryCountLimitReached`? Why do we need to sum them up?

Reply:
Retry count limit can be reached for 1 single fetch call and each fetch call can have multiple events and each demo run can trigger multiple fetch calls. Hence, it can be sum or just the count but that will be setup later when we start posting metrics. I need success to check that we see success fetch calls when the canaries run.

Reply:
yup, sounds good to me